src/baldr/graphreader.cc

```diff
@@ -203,39 +168,22 @@ TileCache* TileCacheFactory::createTileCache(const boost::property_tree::ptree&
 }
 
 // Constructor using separate tile files
-GraphReader::GraphReader(const boost::property_tree::ptree& pt)
-    : tile_hierarchy_(pt.get<std::string>("tile_dir")),
-      tile_extract_(get_extract_instance(pt)),
+GraphReader::GraphReader(const std::shared_ptr<GraphTileStorage>& tile_storage, const boost::property_tree::ptree& pt)
+    : tile_hierarchy_(tile_storage),
       cache_(TileCacheFactory::createTileCache(pt)) {
-  // Reserve cache (based on whether using individual tile files or shared,
-  // mmap'd file
-  cache_->Reserve(tile_extract_->tiles.empty() ? AVERAGE_TILE_SIZE : AVERAGE_MM_TILE_SIZE);
+  // Reserve cache
+  cache_->Reserve(AVERAGE_TILE_SIZE);
```

Comment:
this is a non trivial change here. it means memory mapped tars will reserve tons more memory which is a no go. the whole point of memory mapping is to save memory and this is the main way we run our hosted service. since the `GraphTileStorage` object controls what type of tiles we are loading, maybe we need to add to that interface a method to get the average tile size. then we can add that ternary operator to `GraphTileFsStorage::GetAverageTileSize()` or something like that..